### PR TITLE
Building With Responsive Design : TNW-Website-Clone

### PR DESCRIPTION
**Pull request by DeliceLydia**
This project consists of replicating The Next Web website putting emphasis on how it behaves differently depending on the size of the screen where the website is rendered.

- Setup project dependencies like GitHub Actions and Linters via Microverse Linters Config.

- Used a mobile-first approach when dealing with media queries with 2 breakpoints

- Aligned all elements with float and flex or grid.

### CODE REVIEW

**Comment by DeliceLydia** [comment]
> Try to use shorter commit messages.
> 
> You can include a description where necessary.

noted

File: index.html [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+        <div class="nav-list3">
+          <ul class="nav-section">
+            <li><a href="">LATEST</a></li>
+            <li><a href="">HARD FORK</a></li>
+            <li><a href="">PLUGGED</a></li>
+            <li><a href="">README</a></li>
+            <li><a href="">GROWTH QUARTERS</a></li>
+            <li><a href="">SHIFT</a></li>
+            <li><input type="text" name="search" id="search"></li>
+          </ul>
```
Comment (SafaErden) [inline index.html]: - [x] The page contains a search icon at the rightmost side of the sub-navigation row in the original large screen but the search icon is missing in your version. Please add it.

- [x] Also the navigation and the sub-navigation items should be aligned left but in your version, there is a redundant empty space, please fix your CSS rules to create the same UI with the original page.

image

File: index.html [hunk 2]
```diff
@@ -0,0 +1,533 @@
...
+  <link rel="stylesheet" href="./css/style.css">
+</head>
+
+<body>
+  <header>
+    <nav class="nav">
+      <div class="nav-logo">
+        <a href="index.html"><img src="./images/tnw-red.svg" alt=""></a>
+        <a href=""><i class="fas fa-bars bar fa-2x"></i></a>
+      </div>
```
Comment (SafaErden) [inline index.html]: - [x] In medium view the TWN logo appears at the leftmost part but the navigation menu overlaps it and there is a redundant empty space on the right side of the navigation menu. Please kindly fix it.

image

File: index.html [hunk 3]
```diff
@@ -0,0 +1,533 @@
...
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <script src="https://kit.fontawesome.com/5b1eeceadd.js" crossorigin="anonymous"></script>
+  <title>TNW-Website-Clone</title>
+  <link rel="stylesheet" href="./css/style.css">
+</head>
+
+<body>
+  <header>
+    <nav class="nav">
```
Comment (SafaErden) [inline index.html:14]: - [x] In the original page in the small screen, the Navigation and sub-navigation row changes to a hamburger (bars) icon at the rightmost part (next to seach icon), but in your version, the search icon is missing and the hamburger bars are not aligned right. Please fix these issues.

image

File: index.html [hunk 4]
```diff
@@ -0,0 +1,533 @@
...
+            <h2 class="inside-heading">TikTok says it’s committed to diversity — its history of censorship says
+              otherwise</h2>
+            <span class="small-text">by Bryan Clark</span>
+          </div>
+        </div>
+        <div class="second-img">
+          <div>
+            <img src="./images/hdc-2020-img-1.jpg" alt="" class="main-img img-right">
+          </div>
+          <div>
```
Comment (SafaErden) [inline index.html]: - [x] The second article's title is missing, please add it.

image

File: index.html [hunk 5]
```diff
@@ -0,0 +1,533 @@
...
+      <article class="flex-pictures">
+        <div class="first-img">
+          <img src="./images/tiktok-censorship-lgbtq-796x417.jpg" alt="" class="main-img img-left">
+          <div class="inside">
+            <h5>INSIGHT</h5>
+            <h2 class="inside-heading">TikTok says it’s committed to diversity — its history of censorship says
+              otherwise</h2>
+            <span class="small-text">by Bryan Clark</span>
+          </div>
+        </div>
```
Comment (SafaErden) [inline index.html]: - [x] The top 3 articles section in large screens should contain the TWN logo at the top left corner of the row ([link to page](https://thenextweb.com/)). But the logo is missing in your version. Please kindly add it.

image

File: index.html [hunk 6]
```diff
@@ -0,0 +1,533 @@
...
+        <div class="second-img">
+          <div>
+            <img src="./images/hdc-2020-img-1.jpg" alt="" class="main-img img-right">
+          </div>
+          <div>
+            <img src="./images/remote-work-home-gq-368x276.png" alt="" class="main-img img-right">
+          </div>
+          <div class="inside">
```
Comment (SafaErden) [inline index.html]: - [x] Please add some space between the articles for the small screen view.

image

File: index.html [hunk 7]
```diff
@@ -0,0 +1,533 @@
...
+            <h5>TECH</h5>
+            <h2 class="black inside-heading">5 lessons I’ve learned working from home</h2>
+            <span class="black-small-text">by Tom Popomaronis 11 hours ago</span>
+          </div>
+        </div>
+      </article>
+    </div>
+    <section class="hard-fork">
+      <h2 class="latest-heading">Latest news</h2>
```
Comment (SafaErden) [inline index.html:93]: - [x] There is an empty space on the right side of the latest news section in the medium screen view. Please fix it to create the same UI with the original page.

image

File: index.html [hunk 8]
```diff
@@ -0,0 +1,533 @@
...
+          <div class="col-2">
+            <h4>This AI crunched the numbers to make the perfect TikTok song — but is it though?</h4>
+            <span class="small-text">Thomas Macaulay 1 day ago</span>
+          </div>
+        </article>
+      </div>
+    </section>
+    <div class="flex">
+      <div class="flex-container">
+        <h2 class="section-h2">HARD FORK</h2>
```
Comment (SafaErden) [inline index.html]: - [x] This title shouldn't be all letters capitalized. 

image

File: index.html [hunk 9]
```diff
@@ -0,0 +1,533 @@
...
+            <img src="./images/stonksgates.jpg" alt="" class="main-img">
+          </div>
+          <div class="col-2">
+            <h4>This AI crunched the numbers to make the perfect TikTok song — but is it though?</h4>
+            <span class="small-text">Thomas Macaulay 1 day ago</span>
+          </div>
+        </article>
+      </div>
+    </section>
+    <div class="flex">
```
Comment (SafaErden) [inline index.html:290]: - [x] There is a redundant empty space on the right side of your page in the medium screen view. Please fix your content to fill the full width independent from the screen size.

image

File: index.html [hunk 10]
```diff
@@ -0,0 +1,533 @@
...
+      <div class="flex-container">
```
Comment (SafaErden) [inline index.html:291]: - [x] This section should have 3 category boxes per row in large screens, but in your version, there are 4 boxes. Please kindly fix it.

image

File: index.html [hunk 11]
```diff
@@ -0,0 +1,533 @@
...
+              <li><a href="">Tony Hawk's Pro Skater 1 + 2 is the only good thing this hell year</a></li>
+              <li><a href="">Disney sets a date for Mulan's non-premium streaming debut</a></li>
+              <li><a href="">Baby Yoda returns in The Mandalorian Season 2 this October</a></li>
+              <li><a href="">Netflix is offering a selection of original films and shows for free</a></li>
+            </ul>
+          </div>
+        </div>
+      </div>
+    </div>
+    <div class="hard-fork flex-container">
```
Comment (SafaErden) [inline index.html:565]: - [x] There is a redundant empty space in the medium screen view on the right side. Please fix your content to fill the full width independent from the screen size.

image

File: index.html [hunk 12]
```diff
@@ -0,0 +1,533 @@
...
+          <a href="">ABOUT</a>
+          <a href="">ADVERTISE</a>
+          <a href="">JOBS</a>
+          <a href="">CONTACT</a>
+      </ul>
+    </div>
+    <div class="footer-bottom">
+      <p>&copy; 2006-2020 The Next Web B.V. Made with in Amsterdam.
+        Powered by StackPath</p>
+    </div>
```
Comment (SafaErden) [inline index.html]: - [x] The footer contains TNW logo with a copyright paragraph ([link to page](https://thenextweb.com/)). But the logo is missing in your version. Please add it.

image

File: css/style.css [hunk 13]
```diff
@@ -0,0 +1,649 @@
...
+  display: inline-block;
+  width: 1em;
+}
+
+.latest-list li {
+  margin-top: 6px;
+  margin-left: 15px;
+}
+
+@media (min-width: 320px) {
```
Comment (SafaErden) [inline css/style.css]: - [x] No small breakpoint as it is mobile-first and the base styles are for the mobile screen. ([Screenshot on mobile-first approach](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/responsive_design/mobile_first.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59540217/))

* medium: min-width around 768px
* large: min-width 1024px

File: css/style.css [hunk 14]
```diff
@@ -0,0 +1,649 @@
+/* reset  */
+
+/* base */
+
+/* layout */
+
+/* template */
```
Comment (SafaErden) [inline css/style.css]: - [x] Please remove unnecessary comments.

File: index.html [hunk 15]
```diff
@@ -0,0 +1,533 @@
...
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <script src="https://kit.fontawesome.com/5b1eeceadd.js" crossorigin="anonymous"></script>
+  <title>TNW-Website-Clone</title>
+  <link rel="stylesheet" href="./css/style.css">
+</head>
+
+<body>
+  <header>
```
Comment (SafaErden) [inline index.html:13]: - [x] Instead of wrapping the `nav `tag with `header `tag, you should wrap your `header `tag with the `nav `tag.

**Review by SafaErden — CHANGES_REQUESTED** [review]
## Status: Require changes ♻️

**Hi @DeliceLydia, your project is awesome. 👐 However, we need a little fix before we can proceed. Kindly find the comments below**
##

**REQUIRED CHANGES**

- [x] The Latest Funding section is missing. Please add it as it appears on the original page.

- [x] Make sure that basic CSS bad practices are not used ([article for support ](https://speckyboy.com/good-bad-css-practices)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/section/css_bad_practices_requirement.png) from Pathwright)

* You repeat some of the CSS rules several times in your stylesheet. Please apply DRY CSS rules.

- [ ] The other requested changes are listed below. Please kindly fix them.

##

**NOTES**

Please use the imperative tense for all your commit messages in your upcoming projects.

##

**SUMMARY**

As I mentioned above, your project is almost done. You have some minor issues to solve. Please be an optimist and do not forget that the main purpose is to learn and practice. Do not let the requested changes discourage you 👍

##

*Kindly make the changes and submit another code review when you are ready*

Questions or comments? Reach me on [Twitter](https://twitter.com/safaerden) or [LinkedIn](https://www.linkedin.com/in/safaerden/) 🗨️

Happy coding_ 👨‍💻 

**Safa ERDEN**

File: index.html [hunk 16]
```diff
@@ -0,0 +1,551 @@
...
+        <div class="nav-list3">
+          <ul class="nav-section">
+            <li><a href="">LATEST</a></li>
+            <li><a href="">HARD FORK</a></li>
+            <li><a href="">PLUGGED</a></li>
+            <li><a href="">README</a></li>
+            <li><a href="">GROWTH QUARTERS</a></li>
+            <li><a href="">SHIFT</a></li>
+            <li><input type="text" name="search" id="search"></li>
+          </ul>
```
Comment (teekaytech) [inline index.html]: I can see that you've included the search icon, but it's only showing on small screen size. The requirement says you should implement the search icon for a large screen size. Rendering the icon on small screen size is okay, but not required.

**Large version**
- [x] It contains search icon at the rightmost side of sub-navigation row ([link to page](https://thenextweb.com/))

File: index.html [hunk 17]
```diff
@@ -0,0 +1,551 @@
...
+  <main class="main-container">
+    <div class="first-section">
+      <article class="flex-pictures">
+        <div class="cover-logo"></div>
+        <div class="first-img">
+          <img src="./images/icons8-clown-developer-title-computer-gq-840x628.png" alt="" class="main-img img-left">
+          <div class="inside">
+            <h5>TECH</h5>
+            <h2 class="inside-heading">Why these developer job titles are ridiculous and shouldn’t exist</h2>
+            <span class="small-text">by Bryan Clark</span>
```
Comment (teekaytech) [inline index.html:64]: The last TSE mentioned that you should place the TWN logo at the top left corner of the row. Kindly look at the [main page](https://thenextweb.com/) and add the logo to your version, for medium screen. The box is present, but the TNW wording is missing.

**Medium version changes**
- [x]  TWN logo appears at the leftmost part ([link to page](https://thenextweb.com/))

File: css/style.css [hunk 18]
```diff
@@ -0,0 +1,722 @@
...
+
+/*
```
Comment (teekaytech) [inline css/style.css]: These comments are not necessary. Kindly remove them.

**Review by teekaytech — CHANGES_REQUESTED** [review]
# Changes Required :point_down:

**Hi @DeliceLydia, you have done an amazing job so far :muscle:. But there's a saying which says *it is not over until it is over*. Let's quickly make the changes below before we proceed**. 

* * *

**[ REQUIRED ]**

- [x] The latest funding section aligns 4 articles per row ([link to page](https://thenextweb.com/)) for large screens
- [x] The latest funding section aligns 1 articles per row ([link to page](https://thenextweb.com/)) for medium screen version
- [x] In the latest news section, the large screen size version is having more empty space to the right. Please find a way to justify this space between the left and right margins.
- [x] I pointed out some other required changes below. Please go through them and fix them as appropriate.

Comments:
- Try to be consistent in the way you use units (px, em, %, etc) for your CSS rules. Read more about CSS units and values from this [MDN page](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Values_and_units)
* * *
**In Summary**
We all know that this project is a big one with massive content. Try to observe closely the main website. Everything I pointed out is well placed in there. You are almost there, kindly make the changes carefully and submit another request form as soon as you are done.

**I'm available on github (@teekaytech) and slack (@taofeekolalere)**

**Comment by arslanbisharat** [comment]
### Congratulations APPROVED :clap:
Hello Team! I can see you worked really hard on this project. I reviewed your [project](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59540217/)  and I'm very happy to approve this project with the latest changes you made in this project. :satisfied:
Now you can merge the branches and move forward with your curriculum. 

[OPTIONAL] If you wish to delete your currently working branch please update the Live demo link too so that it works properly. 

_Keep working hard._ 
**Happy Coding**:smiley:
-------------------------------------------------------------------------------------------------------------------
**Regards,
Arslan Bisharat**
[linkedin](https://www.linkedin.com/in/arslanbisharat/) @slack: arslanbisharat

**Review by arslanbisharat — APPROVED** [review]
### Congratulations APPROVED :clap:
Hello Team! I can see you worked really hard on this project. I reviewed your [project](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59540217/)  and I'm very happy to approve this project with the latest changes you made in this project. :satisfied:
Now you can merge the branches and move forward with your curriculum. 

[OPTIONAL] Make your footer exactly the same as the given website by adding width 
[OPTIONAL] If you wish to delete your currently working branch please update the Live demo link too so that it works properly. 

_Keep working hard._ 
**Happy Coding**:smiley:
-------------------------------------------------------------------------------------------------------------------
**Regards,
Arslan Bisharat**
[linkedin](https://www.linkedin.com/in/arslanbisharat/) @slack: arslanbisharat
